test/instructionreader-test.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * instructionreader-test.js
+ */
+"use strict";
+
+import chai from "chai";
+var expect = chai.expect;
+
+/*eslint-disable no-unused-vars*/
+var should = chai.should();
+
+import { prepareRobots, parseInstructions } from "../src/instructionreader.js";
+
+describe("InstructionReader", function() {
+
+    var instructions;
+    var testData;
+
+    var testStack;
+
+    var testRobots;
+
+    before(function() {
+
+        instructions = "5 3\n1 1 E\nRFRFRFRF";
```

Comment:
Did you try making these instance vars? eg. `this.instructions` so you don't need to share them via a closure?